CRM/Core/DomainTokens.php

```diff
@@ -125,6 +126,7 @@ public static function getDomainTokenValues(?int $domainID = NULL, bool $html =
       $email = reset($loc['email']);
       $tokens['phone'] = $phone['phone'] ?? '';
       $tokens['email'] = $email['email'] ?? '';
+      $tokens['base_url'] = (new CRM_Utils_System_Drupal())->getAbsoluteBaseURL();
```

Comment:
We'd need to use a generic function here for cross-CMS support (ie. WordPress, Joomla and different versions of Drupal)

Reply:
Ok, I've made new commit: https://github.com/civicrm/civicrm-core/pull/23852/commits/0f73ad2ace7a7620abd32aa40a858317715f811e